File: Easy/Array/2784. Check if Array is Good.py
# 2784. Check if Array is Good -> python
class Solution:
    def isGood(self, nums: List[int]) -> bool:

        n = len(nums)

        index = [0] * n

        for i in range(0,n,1):
            if( nums[i] > n-1): return False
            index[nums[i]] += nums[i]
            

        for i in range(1,n-1,1):
            if(i != index[i]): return False
        if(index[n-1] == 2*(n-1)) : return True

        return False
        

File: Easy/Array/2784. Check if Array is Good.cpp
//2784. Check if Array is Good ->c++
class Solution {
public:
    bool isGood(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(),nums.end());

        for(int i = 0; i<n;i++){
            if(i != n-1){
                if(i+1 != nums[i]) return false;

            }
            else{
                if(nums[i] != n-1) return false;
            }
        }
        return true;
        
    }
};
